content/blog/using-the-power-of-mdx-with-tina.md

```diff
@@ -0,0 +1,220 @@
+
+```javascript,copy
+const backgroundColor = {
+    warning: "bg-yellow-200",
+    error: "bg-red-600",
+    default: "bg-gray-200"
+}
+
+const textColor = {
+    warning: "text-gray-600",
+    error: "text-white	",
+    default: "text-gray-600"
+}
+
+const Callout = ({callout}) => {
+    const background = backgroundColor[callout.type]; 
+    const color = textColor[callout.type];
+
+return(<div className={`${background} ${color} flex rounded-lg mt-6 p-6`}>
+    <div className="">{callout.text}</div>
+  </div>)
+}
+
+export default Callout;
+
+```
+Before we continue on, let us talk about this component. It is taking a prop named "callout", which provides the type of callout and callout text. The callout type will change the text color and background color.
+
+### Update the tina schema.
+
+In our `schema.ts` when we created our rich-text body section it had a templates array that was empty. We are going to add an object that contains our Callout component, that will drive our editable experience.
+
+The important thing to note is the name needs to match our component name, so our example is Callout. Then the fields will be the parts we want to be editable; so we are going to name them type and text which will be editable by our users.
+
+To make the experience more enjoyable, we are going to add a UI object that holds the defaults. When someone adds our Callout it will automatically have content. Below goes in our templates array.
+
+```typescript,copy
+{
+              name: "Callout",
+              label: "Callout",
+              ui: {
+                defaultItem: {
+                  type: "default",
+                  text: "Lorem ipsum dolor sit amet.",
+                },
+              },
+              fields: [
+                {
+                  name: "type",
+                  label: "Type",
+                  type: "string",
+                  options: ["default", "warning", "error"],
+                },
+                {
+                  name: "text",
+                  label: "Text",
+                  type: "string",
+                },
+              ],
+},
+```
+
+### Registering our component in our front end.
+
+The piece is to register the component with our `TinaMarkdown` . Open up the `[filename].js` file again. First, we need to import our component so add the following to the top of the file:
+
+```javascript,copy
+ import Callout from "../../../components/Callout";
+```
+
+For code clarity, we can create an object that contains all the different Tina-powered components. We are going to pass the props through to our components so we can define that here.
+
+```javascript,copy
+const components = {
+    Callout: (props) => {
+      return <Callout callout={props} />;
+    }};
+```
+Finally, we can update your `TinaMarkdown` component to pass the components for our users to use.
+
+```javascript,copy
+const components = {
+    Callout: (props) => {
+      return <Callout callout={props} />;
+    }};
+```
+
```

Comment:
This should be `<TinaMarkdown content={content} components={components} />`

Reply:
Good catch! I had to redo the code because docs ended up breaking all the code off.